api/v1beta1/azuremanagedcontrolplane_webhook_test.go

```diff
@@ -2683,3 +2770,670 @@ func getAMCPMetaData() metav1.ObjectMeta {
 		Namespace: "default",
 	}
 }
+
+func TestAzureManagedClusterSecurityProfileValidateCreate(t *testing.T) {
+	defer utilfeature.SetFeatureGateDuringTest(t, feature.Gates, capifeature.MachinePool, true)()
+	testsCreate := []struct {
+		name    string
+		amcp    *AzureManagedControlPlane
+		wantErr bool
```

Comment:
I wouldn't hold this PR on this, but generally I'm not a fan of this pattern where we only check whether or not some error occurred. Especially in places like this where there are several different errors that could be returned, it's super easy to think you're returning one but end up getting something else, and a plain bool doesn't protect against that. Instead, I tend to prefer at least defining a sufficiently long substring of the `err.Error()` in the test case, if not the whole error.

Reply:
Ack